src/detection_win.cpp

```diff
@@ -550,3 +551,22 @@ void UpdateDevice(PDEV_BROADCAST_DEVICEINTERFACE pDevInf, WPARAM wParam, DeviceS
 
 	SetEvent(deviceChangedRegisteredEvent);
 }
+
+std::string Utf8Encode(const std::string &str)
+{
+    if (str.empty()) {
+        return std::string();
+    }
+
+    //System default code page to wide character
+    int wstr_size = MultiByteToWideChar(CP_ACP, 0, str.c_str(), -1, NULL, 0);
+    std::wstring wstr_tmp(wstr_size, 0);
+    MultiByteToWideChar(CP_ACP, 0, str.c_str(), -1, &wstr_tmp[0], wstr_size);
```

Comment:
I see this is pretty similar to https://stackoverflow.com/a/6693107/796832 so seems good enough to me